--- draklive-install-1.36-omv3/draklive-install.orig	2013-10-28 13:20:19.217843251 +0000
+++ draklive-install-1.36-omv3/draklive-install	2013-10-28 13:21:48.230847933 +0000
@@ -475,7 +475,7 @@
         }
 
         # Now install the Grub2 itself and create menu configuration for it.
-        $install_err = system("chroot /mnt/install /bin/sh -c 'grub2-install $install_device && grub2-mkconfig -o /boot/grub2/grub.cfg'");
+        $install_err = system("chroot /mnt/install /bin/sh -c 'echo boot=$install_device > /boot/grub2/drakboot.conf && grub2-install $install_device && grub2-mkconfig -o /boot/grub2/grub.cfg'");
         if ($install_err != 0) {
             $install_err = $in->ask_yesorno(N("Error"), N("Failed to install Grub2:") . " $!\n" . N("Try again?"), 1);
             next;
