diff -ur draklive-install-1.31/draklive-install.xsetup draklive-install-1.31.new/draklive-install.xsetup
--- draklive-install-1.31/draklive-install.xsetup	2010-05-24 16:56:31.000000000 +0200
+++ draklive-install-1.31.new/draklive-install.xsetup	2011-08-09 10:06:20.000000000 +0200
@@ -2,6 +2,8 @@
 # to be sourced
 if grep -q '\binstall\b' /proc/cmdline; then
   . /etc/sysconfig/i18n
+  xset -dpms
+  xset s 0 0
   /usr/sbin/draklive-install
   reboot
 fi
