--- draklive-install-1.36-omv3/draklive-install.orig	2013-10-28 20:40:07.209351748 +0000
+++ draklive-install-1.36-omv3/draklive-install	2013-10-28 20:40:14.196352115 +0000
@@ -110,7 +110,7 @@
     my $running_wm = any::running_window_manager();
     my $has_running_wm = to_bool($running_wm);   
     local $::isStandalone = $has_running_wm; # center me if run in xsetup.d script
-    my $w = ugtk2->new(N("OpenMandriva LX Live"));
+    my $w = ugtk2->new(N("OpenMandriva Lx Live"));
     return if ($running_wm eq 'drakx-matchbox-window-manager');
     #ROSAOne-install icon does look like it's openclipart, so stay with it for now
     ugtk2::gtkadd($w->{window},
